uint16 position
int16 velocity